documentation/source/building.rst

```diff
@@ -5,143 +5,167 @@ Build options and Environment Variables
-
-CUSTOM_COMPILE_DEPS
-~~~~~~~~~~~~~~~~~~~
-
-Use to add additional dependencies to the compilation target; useful for defining additional rules to run pre-compilation or if the compilation phase depends on files other than the RTL sources listed in **VERILOG_SOURCES** or **VHDL_SOURCES**.
-
-CUSTOM_SIM_DEPS
-~~~~~~~~~~~~~~~
-
-Use to add additional dependencies to the simulation target.
-
-COCOTB_NVC_TRACE
-~~~~~~~~~~~~~~~~
-
-Set this to 1 to enable display VHPI trace when using nvc VHDL simulator.
-
-SIM_BUILD
-~~~~~~~~~
-
-Use to define a scratch directory for use by the simulator. The path is relative to the Makefile location. If not provided, the default scratch directory is "sim_build".
-
+.. glossary::
+
+    ``GUI``
+      Set this to 1 to enable the GUI mode in the simulator (if supported).
+      
+    ``SIM``
+      Selects which simulator Makefile to use.  Attempts to include a simulator specific makefile from :file:`cocotb/makefiles/makefile.$(SIM)`
+      
+    ``VERILOG_SOURCES``
+      A list of the Verilog source files to include.
+      
+    ``VHDL_SOURCES``
+      A list of the VHDL source files to include.
+      
+    ``COMPILE_ARGS``
+      Any arguments or flags to pass to the compile stage of the simulation. Only applies to simulators with a separate compilation stage (currently Icarus and VCS).
+      
+    ``SIM_ARGS``
+      Any arguments or flags to pass to the execution of the compiled simulation.  Only applies to simulators with a separate compilation stage (currently Icarus, VCS and GHDL).
+      
+    ``EXTRA_ARGS``
+      Passed to both the compile and execute phases of simulators with two rules, or passed to the single compile and run command for simulators which don't have a distinct compilation stage.
+
+.. comment
+..    ``GENERICS``
+..      Override the generics in the design.
+..      Note that capabilities may vary between simulators, e.g. whether you can only set generics of certain types, or only set toplevel generics, and, if hierarchical specifications are supported, in what format to specify them.
+    
+    ``CUSTOM_COMPILE_DEPS``
+      Use to add additional dependencies to the compilation target; useful for defining additional rules to run pre-compilation or if the compilation phase depends on files other than the RTL sources listed in :term:`VERILOG_SOURCES` or :term:`VHDL_SOURCES`.
+      
+    ``CUSTOM_SIM_DEPS``
+      Use to add additional dependencies to the simulation target.
+      
+    ``COCOTB_NVC_TRACE``
+      Set this to 1 to enable display of VHPI traces when using the nvc VHDL simulator.
+      
+    ``SIM_BUILD``
+      Use to define a scratch directory for use by the simulator. The path is relative to the Makefile location.
+      If not provided, the default scratch directory is :file:`sim_build`.
+
```

Comment:
What's the reason for having these variables commented? If they are valid, I'd make them visible (and possibly mark as deprecated or partially working). Otherwise can you remove them completely?

Reply:
Only ``GENERICS`` is commented out, and it's because of the outcome of #682, where (maybe) the consensus is that we cannot really support it for all simulators, so should be removed completely.
How do we get actual consensus there?

Reply:
I have taken ``GENERICS`` out now; see also #692 where the variable is removed from the makefiles.